api/python/t4/api.py

```diff
@@ -449,3 +450,121 @@ def config(*autoconfig_url, **config_values):
         write_yaml(local_config, CONFIG_PATH, keep_backup=True)
 
     return HeliumConfig(CONFIG_PATH, local_config)
+
+def create_role(name, arn):
+    """
+    Create a new role in your registry. Admins only.
+
+    Required Parameters:
+        name(string): name of role to create
+        arn(string): ARN of IAM role to associate with the Quilt role you are creating
+    """
+    session = get_session()
+    response = session.post(
+        "{url}/api/roles".format(
+            url=get_registry_url()
+            ),
+            data=json.dumps({
+                'name': name,
+                'arn': arn
+            })
+        )
+    if not response.ok:
+        message = response.json()['message']
+        raise QuiltException("Error creating role: " + message)
+
+    return response.json()
+
+def edit_role(role_id, new_name=None, new_arn=None):
+    """
+    Edit an existing role in your registry. Admins only.
+
+    Required parameters:
+        role_id(string): ID of role you want to operate on.
+
+    Optional paramters:
+        new_name(string): new name for role
+        new_arn(string): new ARN for IAM role attached to Quilt role
+    """
+    session = get_session()
+    old_data = get_role(role_id)
+    data = {}
+    data['name'] = new_name or old_data['name']
+    data['arn'] = new_arn or old_data['arn']
+
+    response = session.put(
+        "{url}/api/roles/{role_id}".format(
+            url=get_registry_url(),
+            role_id=role_id
+            ),
+            data=data
+        )
+    if not response.ok:
+        raise QuiltException("Failed to edit role")
+
+    return response.json()
+
+def delete_role(role_id):
+    """
+    Delete a role in your registry. Admins only.
+
+    Required parameters:
+        role_id(string): ID of role you want to delete.
+    """
+    session = get_session()
+    response = session.delete(
+        "{url}/api/roles/{role_id}".format(
+            url=get_registry_url(),
+            role_id=role_id
+            )
+        )
+    if not response.ok:
+        raise QuiltException("Failed to delete role")
+
+def get_role(role_id):
+    """
+    Get info on a role based on its ID. Admins only.
+
+    Required parameters:
+        role_id(string): ID of role you want to get details on.
+    """
+    session = get_session()
+    response = session.get(
+        "{url}/api/roles/{role_id}".format(
+            url=get_registry_url(),
+            role_id=role_id
+            )
+        )
+    if not response.ok:
+        raise QuiltException("Role not found")
+
+    return response.json()
+
+def list_roles():
+    """
+    List configured roles. Admins only.
+    """
+    session = get_session()
+    response = session.get(
+        "{url}/api/roles".format(
+            url=get_registry_url()
+        ))
```

Comment:
Any reason why you dropped     
```
if not response.ok:
        raise QuiltException("...")
```

here and don't have it inset_role?

Reply:
the session raises an exception on its own if the response has a 4xx or 5xx code

Reply:
Cool, thanks for clarifying